### PR TITLE
Added Prisma and /public API

### PR DESCRIPTION
**Pull request by BolverBlitz**

### CODE REVIEW

**Review by DarkIntaqt — APPROVED** [review]
LGTM

**Review by DarkIntaqt — APPROVED** [review]
LGTM

File: landingpage/src/Components/Events.astro [hunk 1]
```diff
@@ -4,34 +4,29 @@ import Heading from "./Heading.astro";
...
+  date_apply: string;
+  min_group: string;
+  visibility: number;
+  state: number;
+  creator_user_id: number;
+};
+
+const events: EventDTO[] =
+  ((await (
+    await fetch(`${import.meta.env.API_URL}/api/v1/public/events`)
```
Comment (BolverBlitz) [inline landingpage/src/Components/Events.astro:28]: This is failing in the building phase.

```cmd
 generating static routes
02:39:54 ▶ src/pages/en.astro
02:39:54   └─ /en/index.htmlFailed to parse URL from "http://localhost/api/v1/public/events
  Stack trace:
    at node:internal/deps/undici/undici:12502:13
  Caused by:
  Invalid URL
    at new URL (node:internal/url:797:36)
    at fetch (node:internal/deps/undici/undici:9662:25)
    at value (node:internal/bootstrap/web/exposed-window-or-worker:72:12)
    at AstroComponentInstance.Events [as factory] (file:///C:/Users/1515/Documents/GitHub/EBG-Webpage/landingpage/dist/chunks/astro/server_D4nMDkhF.mjs:23:12)
    at AstroComponentInstance.render (file:///C:/Users/1515/Documents/GitHub/EBG-Webpage/landingpage/dist/chunks/astro/server_D4nMDkhF.mjs:876:36)
```
    
When this is happening, the API is returning 
```json
[]
```

Comment (BolverBlitz) [reply landingpage/src/Components/Events.astro:28]: It also fails with a actual response 

```json
[{"id":2,"puuid":"3c36f80e-0c04-4a85-a9e4-95e718cf55b7","type":1,"name":"Test","description":"ABCDEFG","avatar_url":"","color":"orange","location_address":"Test","date_start":"2024-08-31T22:00:00.000Z","date_end":"2025-03-01T23:00:00.000Z","date_created":"2024-08-07T00:57:25.419Z","date_apply":"2024-08-07T22:00:00.000Z","min_group":"user","visibility":1,"state":0,"creator_user_id":1}]
```

**Comment by DarkIntaqt** [comment]
Gimme a day or two, I am rewriting the data fetching to fix that issue

**Pull request merged**